diff --git a/collectors/cgroups.plugin/cgroup-name.sh.in b/collectors/cgroups.plugin/cgroup-name.sh.in
index 3aebe2bf..917d160a 100755
--- a/collectors/cgroups.plugin/cgroup-name.sh.in
+++ b/collectors/cgroups.plugin/cgroup-name.sh.in
@@ -139,6 +139,10 @@ if [ -z "${NAME}" ]; then
 		# systemd-nspawn
 		NAME="$(echo "${CGROUP}" | sed 's/.*machine.slice[_\/]\(.*\)\.service/\1/g')"
 
+	elif [[ "${CGROUP}" =~ machine.slice[_/].*\.scope ]]; then
+		# machinectl
+		NAME="$(systemd-escape -u $(echo ${CGROUP} | sed 's/.*machine.slice[_\/]machine-\(.*\)\.scope/\1/g' | tr '/' '\\'))"
+
 	elif [[ ${CGROUP} =~ machine.slice_machine.*-qemu ]]; then
 		# libvirtd / qemu virtual machines
 		# NAME="$(echo ${CGROUP} | sed 's/machine.slice_machine.*-qemu//; s/\/x2d//; s/\/x2d/\-/g; s/\.scope//g')"
