diff --git a/collectors/cgroups.plugin/cgroup-name.sh.in b/collectors/cgroups.plugin/cgroup-name.sh.in
index 53696a4b..6ecf39ce 100755
--- a/collectors/cgroups.plugin/cgroup-name.sh.in
+++ b/collectors/cgroups.plugin/cgroup-name.sh.in
@@ -152,6 +152,12 @@ if [ -z "${NAME}" ]
 
         NAME="$(echo ${CGROUP} | sed 's/.*machine.slice[_\/]\(.*\)\.service/\1/g')"
 
+    elif [[ "${CGROUP}" =~ machine.slice[_/].*\.scope ]]
+        then
+        # machinectl
+
+        NAME="$(systemd-escape -u $(echo ${CGROUP} | sed 's/.*machine.slice[_\/]machine-\(.*\)\.scope/\1/g' | tr '/' '\\'))"
+
     elif [[ "${CGROUP}" =~ machine.slice_machine.*-qemu ]]
         then
         # libvirtd / qemu virtual machines
