diff --git a/collectors/python.d.plugin/hddtemp/hddtemp.chart.py b/collectors/python.d.plugin/hddtemp/hddtemp.chart.py
index dea70117..6c8bcc03 100644
--- a/collectors/python.d.plugin/hddtemp/hddtemp.chart.py
+++ b/collectors/python.d.plugin/hddtemp/hddtemp.chart.py
@@ -27,7 +27,7 @@ RE = re.compile(r'\/dev\/([^|]+)\|([^|]+)\|([0-9]+|SLP|UNK)\|')
 class Disk:
     def __init__(self, id_, name, temp):
         self.id = id_.split('/')[-1]
-        self.name = name.replace(' ', '_')
+        self.name = self.id.replace(' ', '_')
         self.temp = temp if temp.isdigit() else 0
 
     def __repr__(self):
